spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/RuntimeModulesController.java

```diff
@@ -0,0 +1,194 @@
+@RequestMapping("/runtime/modules")
+@ExposesResourceFor(ModuleStatusResource.class)
+public class RuntimeModulesController {
+
+	private static final Comparator<? super ModuleInstanceStatus> INSTANCE_SORTER = new Comparator<ModuleInstanceStatus>() {
+		@Override
+		public int compare(ModuleInstanceStatus i1, ModuleInstanceStatus i2) {
+			return i1.getId().compareTo(i2.getId());
+		}
+	};
+
+	private final Collection<ModuleDeployer> moduleDeployers;
+
+	private final ResourceAssembler<ModuleStatus, ModuleStatusResource> statusAssembler = new Assembler();
+
+	@Autowired
+	public RuntimeModulesController(Collection<ModuleDeployer> moduleDeployers) {
+		this.moduleDeployers = new HashSet<>(moduleDeployers);
+	}
+
+	@RequestMapping
+	public PagedResources<ModuleStatusResource> list(PagedResourcesAssembler<ModuleStatus> assembler) {
+		List<ModuleStatus> values = new ArrayList<>();
+		for (ModuleDeployer moduleDeployer : moduleDeployers) {
+			values.addAll(moduleDeployer.status().values());
+		}
+		Collections.sort(values, new Comparator<ModuleStatus>() {
+			@Override
+			public int compare(ModuleStatus o1, ModuleStatus o2) {
+				return o1.getModuleDeploymentId().toString().compareTo(o2.getModuleDeploymentId().toString());
+			}
+		});
+		return assembler.toResource(new PageImpl<>(values), statusAssembler);
+	}
+
+	@RequestMapping("/{id}")
+	public ModuleStatusResource display(@PathVariable String id) {
+		ModuleDeploymentId moduleDeploymentId = ModuleDeploymentId.parse(id);
+		for (ModuleDeployer moduleDeployer : moduleDeployers) {
+			ModuleStatus status = moduleDeployer.status(moduleDeploymentId);
+			if (status != null) {
+				return statusAssembler.toResource(status);
+			}
+		}
+		throw new ResourceNotFoundException();
+	}
+
+	private class Assembler extends ResourceAssemblerSupport<ModuleStatus, ModuleStatusResource> {
+
+		public Assembler() {
+			super(RuntimeModulesController.class, ModuleStatusResource.class);
+		}
+
+		@Override
+		public ModuleStatusResource toResource(ModuleStatus entity) {
+			return createResourceWithId(entity.getModuleDeploymentId(), entity);
+		}
+
+		@Override
+		protected ModuleStatusResource instantiateResource(ModuleStatus entity) {
+			ModuleStatusResource resource = new ModuleStatusResource(entity.getModuleDeploymentId().toString(), entity.getState().name());
+			List<ModuleInstanceStatusResource> instanceStatusResources = new ArrayList<>();
+			InstanceAssembler instanceAssembler = new InstanceAssembler(entity);
+			List<ModuleInstanceStatus> instanceStatuses = new ArrayList<>(entity.getInstances().values());
+			Collections.sort(instanceStatuses, INSTANCE_SORTER);
+			for (ModuleInstanceStatus moduleInstanceStatus : instanceStatuses) {
+				instanceStatusResources.add(instanceAssembler.toResource(moduleInstanceStatus));
+			}
+			resource.setInstances(new Resources<>(instanceStatusResources));
+			return resource;
+		}
+	}
+
+	@RestController
+	@RequestMapping("/runtime/modules/{moduleId}/instances")
```

Comment:
Not the scope of this PR. But, it would be nice to use these REST endpoints from the shell commands as well.